From 69ef32a841bdbd19e3f076b6e9f95f80b5a3271a Mon Sep 17 00:00:00 2001
From: "Langer, Thomas" <tlanger@maxlinear.com>
Date: Tue, 11 May 2021 12:47:55 +0200
Subject: [PATCH] PONRTSYS-8555: Remove DP_TX_CB_LOCK

as the initial use (locking list of tx_cb functions) was already removed
and the mib_counters are using a different lock now.
---
 drivers/net/datapath/dpm/gswip31/datapath_misc.c | 1 -
 drivers/net/datapath/dpm/gswip31/datapath_misc.h | 5 -----
 2 files changed, 6 deletions(-)

diff --git a/drivers/net/datapath/dpm/gswip31/datapath_misc.c b/drivers/net/datapath/dpm/gswip31/datapath_misc.c
index 982d0e190173..2b8ab4d4e2e5 100644
--- a/drivers/net/datapath/dpm/gswip31/datapath_misc.c
+++ b/drivers/net/datapath/dpm/gswip31/datapath_misc.c
@@ -1402,7 +1402,6 @@ static int dp_platform_set(int inst, u32 flag)
 			return DP_FAILURE;
 
 		priv = (struct hal_priv *)dp_prop->priv_hal;
-		DP_TX_CB_LOCK_INIT(&priv->cb_lock);
 
 		/* Drop Q is currently mapped to 255 */
 		priv->ppv4_drop_q = MAX_QUEUE - 1;
diff --git a/drivers/net/datapath/dpm/gswip31/datapath_misc.h b/drivers/net/datapath/dpm/gswip31/datapath_misc.h
index bfd5ae3b09ba..90136be490f7 100644
--- a/drivers/net/datapath/dpm/gswip31/datapath_misc.h
+++ b/drivers/net/datapath/dpm/gswip31/datapath_misc.h
@@ -199,10 +199,6 @@ struct resv_info {
 	struct resv_sch *resv_sched; /*!< reserved schedulers info */
 };
 
-#define DP_TX_CB_LOCK_T		spinlock_t
-#define DP_TX_CB_LOCK		spin_lock_bh
-#define DP_TX_CB_UNLOCK		spin_unlock_bh
-#define DP_TX_CB_LOCK_INIT	spin_lock_init
 struct dp_spl_cfg;
 struct pp_qos_dev;
 struct hal_priv {
@@ -225,7 +221,6 @@ struct hal_priv {
 							 *       <-> DPID
 							 */
 	struct dp_spl_cfg *spl;
-	DP_TX_CB_LOCK_T cb_lock;
 	spinlock_t qos_lock;
 };
 
